src/gwt/src/org/rstudio/studio/client/workbench/MainWindowObject.java

```diff
@@ -76,16 +75,14 @@ private static final native boolean hasImpl(String key, JavaScriptObject object)
       return object.hasOwnProperty(key);
    }-*/;
    
-   private static final native Window getMainWindow() /*-{
-      var wnd = $wnd;
-      while (wnd.opener != null)
-         wnd = wnd.opener;
-      return wnd;
+   private static final native Window getRStudioMainWindow() /*-{
+      for (var wnd = $wnd; wnd != null; wnd = wnd.opener)
+         if (!!wnd.$RStudio)
+            return wnd;
+      throw "Failed to discover RStudio main window";
```

Comment:
What is the consequence of this exception getting thrown (i.e. where will it be caught/handled)? The previous behavior was to use `$wnd` as the main window in the absence of evidence to the contrary, so perhaps we should preserve that for the patch release.